### PR TITLE
Add new methods for Invoice resource

### PR DESCRIPTION
**Pull request by remi-stripe**
cc @stripe/api-libraries

### CODE REVIEW

File: src/main/java/com/stripe/model/Invoice.java [hunk 1]
```diff
@@ -129,6 +138,68 @@ public static InvoiceCollection list(Map<String, Object> params, RequestOptions
   }
   // </editor-fold>
 
+  // <editor-fold desc="finalize">
+  /**
+   * Finalize an invoice.
+   */
+  public Invoice finalize() throws StripeException {
```
[the remark this answers is not in the document]
Comment (ob-stripe) [reply src/main/java/com/stripe/model/Invoice.java]: We definitely should not -- `finalize` is called by the JVM when the object is GC'd.

**Comment by remi-stripe** [comment]
@ob-stripe the build failure seems related to `oraclejdk10` being deprecated so it likely just needs to be removed but waiting for your 👍 on that one

**Comment by remi-stripe** [comment]
@brandur-stripe re-assigning to you as this one had conflicts and I rebased on the latest master.

**Comment by brandur-stripe** [comment]
LGTM.

**Pull request merged**